src/components/utils/PreviewedLink.tsx

```diff
@@ -0,0 +1,153 @@
+  } catch {
+    return '';
+  }
+};
+
+const fetchOgpContents = async (url: URL): Promise<OgpContents | null> => {
+  const whiteList = ['www3.nhk.or.jp'];
+  if (whiteList.includes(url.host)) {
+    const res = await fetch(url, { headers: { Accept: 'text/html' } });
+    const text = await res.text();
+    const el = new DOMParser().parseFromString(text, 'text/html');
+    const ogs: { [index: string]: string } = {};
+    Array.from(el.head.querySelectorAll('meta'))
+      .filter((v) => v.getAttribute('property') != null && v.getAttribute('content') != null)
+      .forEach((v) => {
+        const property = v.getAttribute('property');
+        const content = v.getAttribute('content');
+        if (property != null && content != null) {
+          ogs[property] = content;
+        }
+      });
+
+    if (ogs['og:image'] && ogs['og:title'] && ogs['og:description']) {
+      const contents: OgpContents = {
+        title: ogs['og:title'],
+        description: ogs['og:description'],
+        image: new URL(ogs['og:image']),
+        url,
+      };
+      return contents;
+    }
+  }
+  return null;
+};
+
+const PreviewedLink: Component<SafeLinkProps> = (props) => {
+  let twitterRef: HTMLQuoteElement | undefined;
+
+  const initialOgp: OgpContents = { title: '', description: '', image: null, url: null };
+  const [getOgpContents, setOgpContents] = createSignal(initialOgp);
+
+  const isSafe = () => {
+    try {
+      const url = new URL(props.href.toString());
+      return url.protocol === 'https:' || url.protocol === 'http:';
+    } catch {
+      return false;
+    }
+  };
+
+  const updateOgpContents = async () => {
+    const ogp = await fetchOgpContents(new URL(props.href.toString()));
+    if (ogp != null) {
+      setOgpContents(ogp);
+    }
+  };
+
+  createEffect(() => {
+    if (isTwitterUrl(props.href.toString())) {
+      window.twttr?.widgets?.load(twitterRef);
+    }
+  });
+
+  onMount(() => {
+    updateOgpContents()
+      .then(() => {})
+      .catch(() => {});
+  });
+
+  return (
+    <Show when={isSafe()} fallback={props.href}>
+      <Switch fallback={props.href}>
+        <Match when={isTwitterUrl(props.href.toString())}>
+          <blockquote class="twitter-tweet" ref={twitterRef}>
+            <a
+              class={props.class}
+              href={twitterUrl(props.href.toString())}
+              target="_blank"
+              rel="noreferrer noopener"
+            >
+              {twitterUrl(props.href.toString())}
+            </a>
+          </blockquote>
+        </Match>
+        <Match when={isYoutubeVideoUrl(props.href.toString())}>
+          <div class="my-2 aspect-video w-full">
+            <iframe title="YouTube" class="h-full w-full" src={youtubeUrl(props.href.toString())} />
+          </div>
+        </Match>
+        <Match when={getOgpContents().url === null}>
+          <SafeLink class={props.class} href={props.href} />
+        </Match>
+        <Match when={getOgpContents().url !== null}>
```

Comment:
必須ではないので、残してあれば私の方で直します

keyedを使うと、引数がnullやundefinedでない場合のみ関数が実行されます。

```suggestion
        <Match when={getOgpContents()} keyed>
          ｛(ogp) => (中身)}
        </Match>
```